spec/services/work_packages/shared/shared_examples_days.rb

```diff
@@ -178,40 +178,56 @@
 end
 
 RSpec.shared_examples "add lag to a date" do
-  describe "#add_lag" do
+  describe "#with_lag" do
     saturday_2022_07_30 = Date.new(2022, 7, 30)
     sunday_2022_07_31 = Date.new(2022, 7, 31)
     monday_2022_08_01 = Date.new(2022, 8, 1)
 
-    it "returns the soonest date after the given date where the number of working days in between equals the given lag" do
-      expect(subject.add_lag(sunday_2022_07_31, 0)).to eq(monday_2022_08_01)
-      expect(subject.add_lag(sunday_2022_07_31, 1)).to eq(Date.new(2022, 8, 2))
-      expect(subject.add_lag(sunday_2022_07_31, 5)).to eq(Date.new(2022, 8, 6))
-      expect(subject.add_lag(sunday_2022_07_31, 6)).to eq(Date.new(2022, 8, 7))
+    context "when lag is positive" do
+      it "returns the soonest date after the given date where the number of working days in between equals the given lag" do
+        expect(subject.with_lag(sunday_2022_07_31, 0)).to eq(monday_2022_08_01)
+        expect(subject.with_lag(sunday_2022_07_31, 1)).to eq(Date.new(2022, 8, 2))
+        expect(subject.with_lag(sunday_2022_07_31, 5)).to eq(Date.new(2022, 8, 6))
+        expect(subject.with_lag(sunday_2022_07_31, 6)).to eq(Date.new(2022, 8, 7))
+      end
     end
 
-    it "returns the day after the given date when lag is negative (like lag = 0)" do
-      expect(subject.add_lag(sunday_2022_07_31, -100)).to eq(monday_2022_08_01)
-      expect(subject.add_lag(sunday_2022_07_31, -1)).to eq(monday_2022_08_01)
-      expect(subject.add_lag(sunday_2022_07_31, 0)).to eq(monday_2022_08_01)
+    context "when lag is negative" do
+      it "returns the latest day before the given date where the number of working days between " \
+         "both dates, inclusive, is equal to the given lag (but negative)" do
+        expect(subject.with_lag(saturday_2022_07_30, -1)).to eq(saturday_2022_07_30)
+
+        expect(subject.with_lag(sunday_2022_07_31, -100)).to eq(sunday_2022_07_31 - 99.days)
+        expect(subject.with_lag(sunday_2022_07_31, -1)).to eq(sunday_2022_07_31)
+        expect(subject.with_lag(sunday_2022_07_31, 0)).to eq(monday_2022_08_01)
+      end
     end
 
     it "works with big lag value like 100_000" do
       # Ensure implementation is not recursive and won't fail with SystemStackError: stack level too deep
-      expect { subject.add_lag(sunday_2022_07_31, 100_000) }
+      expect { subject.with_lag(sunday_2022_07_31, 100_000) }
         .not_to raise_error
     end
 
     context "with weekend days (Saturday and Sunday)", :weekend_saturday_sunday do
```

Comment:
I find this context description ambiguous - I would expect _**with** weekend days_ to include Saturday and Sunday as working days,

Reply:
Good point. Updated to "with Saturday and Sunday being non-working days" as "weekend" has different meanings depending on your country / culture.